scipio/examples/hello_world.rs

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the
+// MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2020 Datadog, Inc.
+//
+use futures::future::join_all;
+use scipio::{Local, LocalExecutor};
+
+async fn hello() {
+    let mut tasks = vec![];
+    for t in 0..5 {
+        tasks.push(Local::local(async move {
+            println!("{}: Hello {} ...", Local::id(), t);
+            Local::later().await;
+            println!("{}: ... {} World!", Local::id(), t);
+        }));
+    }
+    join_all(tasks).await;
+}
+
+fn main() {
+    // There are two ways to create an executor, demonstrated in this example.
+    //
+    // We can create it in the current thread, and run it separately later...
+    let ex = LocalExecutor::new(Some(0)).unwrap();
```

Comment:
`unwrap()` in examples is unfortunate, because you end up under-designing API error handling. This should be trivial to fix to return the error, no?

Reply:
Internals has the following code example:

```rust
let local_ex = LocalExecutor::new(None).expect("failed to create local executor");
local_ex.run(async {
     Timer::new(std::time::Duration::from_millis(100)).await;
     println!("Hello world!");
});
```

Reply:
What do you propose? In examples usually you don't have context do do any error handling. I see a lot of other libraries using unwrap.

Reply:
expect is just unwrap() with an error message. Both panics, and I don't consider either to be proper error handling.
Unwrap is just less verbose.

If there is a preference towards `expect` I can use it, but I don't see a big difference

Reply:
I was not arguing for `expect()` either -- I was arguing for _returning_ the error from `main()` -- e.g. `fn main() -> Result<(), std::io::Error>`.

Reply:
would just using `?` suffice ?

I think we lack context to do anything more complex than that in an example. 
(I am also not sure we can do `?` in examples, but assuming we can)

Reply:
Yup, that was my suggestion: change the return type of `main` and propagate the error with `?`.

Reply:
<facepalm>

you are talking about the hello world example.

I thought you were talking about the examples on top of the functions. Sorry, that's my bad